analyzer-core/src/main/java/ffb/analyzer/core/web/request/BaseEntityEnclosingRequestBuilder.java

```diff
@@ -2,11 +2,30 @@
 
 import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
 
+/**
+ * Common interface to describe a builder class to construct a request with a request body.
+ * @param <T> Type of request to be built.
+ */
 public interface BaseEntityEnclosingRequestBuilder<T extends HttpEntityEnclosingRequestBase> {
 
+    /**
+     * Adds a URL to the request.
+     * @param url URL to send the request to.
+     * @return Reference to the builder class.
+     */
     BaseEntityEnclosingRequestBuilder<T> addUrl(String url);
 
+    /**
+     * Adds content to the request body.
+     * @param key The parameter name in the body.
+     * @param value The value for the parameter.
+     * @return Reference to the builder class.
+     */
     BaseEntityEnclosingRequestBuilder<T> addRequestBodyParameter(String key, String value);
 
+    /**
+     * Builds the request.
```

Comment:
I think we're fine to leave them off if we're good about naming our methods. The benefit is that for obscurely named methods, we can quickly glance at the documentation to get a sense of what the method is supposed to do.

I don't think we'll be able to keep the `@returns` in our Checkstyle if we disable method documentation, but I'll take a look.

Reply:
Ok here's my proposal for method java docs:

Only requirement is the `@returns` and the `@throws`, but you should add documentation if you feel there's something in the method that's worth calling out explicitly like assumptions or other important information.

Thoughts?

Reply:
Yes, I like that approach. Were you able to find how to handle that in the checkstyle, or will it require creating blank java docs for methods that don't return/throw anything?

Reply:
I don't think our Checkstyle will throw errors anymore if java docs are missing since in the majority of cases, we shouldn't need them.

If the java docs are present though, there needs to be a `@return` and `@throws` if applicable.